keel-api/src/test/kotlin/com/netflix/spinnaker/keel/integration/AuthPropagationTests.kt

```diff
@@ -0,0 +1,88 @@
+package com.netflix.spinnaker.keel.integration
+
+import com.netflix.spinnaker.keel.KeelApplication
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.clouddriver.model.Network
+import com.netflix.spinnaker.security.AuthenticatedRequest.Header.XSpinnakerAnonymous
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import kotlinx.coroutines.runBlocking
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.MOCK
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import strikt.api.expect
+import strikt.api.expectThat
+import strikt.assertions.isEmpty
+import strikt.assertions.isEqualTo
+import strikt.assertions.isNotNull
+
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(
+  classes = [KeelApplication::class],
+  properties = [
+    "clouddriver.enabled=true",
+    "clouddriver.base-url=http://localhost:8080"
+  ],
+  webEnvironment = MOCK
+)
+internal class AuthPropagationTests : JUnit5Minutests {
+
+  @Autowired
+  lateinit var cloudDriverService: CloudDriverService
+
+  data class Fixture(
+    private val cloudDriverService: CloudDriverService
+  ) {
+    val server = MockWebServer()
+
+    private var _listNetworksResults: Map<String, Set<Network>>? = null
+    val listNetworksResults: Map<String, Set<Network>>
+      get() = checkNotNull(_listNetworksResults) { "You need to actually make a call first" }
+
+    fun listNetworks() {
+      _listNetworksResults = runBlocking {
+        cloudDriverService.listNetworks().await()
+      }
+    }
+  }
+
+  fun tests() = rootContext<Fixture> {
+    fixture {
+      Fixture(cloudDriverService)
+    }
+
+    before {
+      server.start(8080)
+    }
+
+    after {
+      server.shutdown()
+    }
+
+    context("a call to Clouddriver") {
+      before {
+        server.enqueue(MockResponse().setBody("{}"))
+
+        listNetworks()
+      }
+
+      test("calls the endpoint") {
+        expect {
+          that(server.requestCount).isEqualTo(1)
+          that(listNetworksResults).isEmpty()
+        }
+      }
+
+      test("propagates $XSpinnakerAnonymous header") {
+        expectThat(server.takeRequest())
+          .get { getHeader(XSpinnakerAnonymous) }
+          .isNotNull()
+          .isEqualTo("anonymous")
```

Comment:
Test that the `X-SPINNAKER-ANONYMOUS` header arrives at the server (spoiler, it doesn't)